datasets/es/mayors_councillors/crawler.py

```diff
@@ -0,0 +1,118 @@
+from rigour.mime.types import XLSX
+from openpyxl import load_workbook
+from typing import Dict
+
+from zavod import Context, helpers as h
+from zavod.stateful.positions import categorise
+
+IGNORE = [
+    "column_0",
+    "column_1",
+    "column_2",
+    "autonomous_community",
+    "ine_code",
+    "column_11",
+    "column_12",
+    "column_13",
+    "column_14",
+    "column_15",
+    "column_16",
+    "column_17",
+]
+
+
+def crawl_item(context: Context, row: Dict[str, str]):
+    name = row.pop("name")
+    province = row.pop("province")
+    municipality = row.pop("municipality")
+    first_last_name = row.pop("last_name", None)
+    second_last_name = row.pop("second_last_name", None)
+    position = row.pop("position", None)
+
+    if not name or not province or not municipality:
+        context.log.warning("Missing required fields", row=row)
+        return
+
+    pep = context.make("Person")
+    pep.id = context.make_id(name, province, municipality)
+    if first_last_name or second_last_name:
+        last_name = f"{first_last_name} {second_last_name}"
+        h.apply_name(pep, first_name=name, last_name=last_name)
+    else:
+        pep.add("name", name)
+    pep.add("political", row.pop("party"))
+    position = h.make_position(
+        context,
+        # Positions are available for the current officials; historical data lists only mayors
+        position if position else "Mayor",
```

Comment:
we usually do something like `f"{role} of {municipalities[municipality_code].get('municipality.long_name')}"` or `f"Mayor of {municipality}"`

So that'd be the muni and province in the position name, in case munis in different provinces have the same name.

Are there any positions other than mayor or councillor? Otherwise I think you can just translate both of them

Reply:
you can also add gov.muni to everything and add gov.had to mayor and gov.legislative to councillors if we're already doing logic on the position name.